builders/Dockerfile.mingw64

```diff
@@ -0,0 +1,121 @@
+        -Dscriptlet-shell='/bin/bash' \
+        -Ddoc='disabled' \
+        -Ddoxygen='disabled' \
+        -Ddoc='disabled' \
+        -Di18n=false \
+        build && \
+    ninja -C build && \
+    ninja -C build install && \
+    ln -s /usr/local/pacman/bin/* /usr/local/bin/
+
+COPY mingw64/etc/ /windows/etc/
+
+# setup pacman-key
+RUN set -ex && \
+    # download MSYS2 keyring
+    mkdir -p /usr/local/pacman/share/pacman/keyrings/ && \
+    wget --no-verbose "https://repo.msys2.org/msys/x86_64/${MSYS2_KEYRING_PKG}" && \
+    echo "${MSYS2_KEYRING_PKG_SHA256} *${MSYS2_KEYRING_PKG}" | sha256sum --check --strict - && \
+    tar -x -C /usr/local/pacman/share/pacman/keyrings/ -f "${MSYS2_KEYRING_PKG}" --strip-components 4 usr && \
+    # initialize keyring
+    pacman-key --init && \
+    pacman-key --populate msys2
+
+
+# Main image
+FROM debian:bullseye
+
+LABEL org.opencontainers.image.title="Geany-Mingw-w64-CI"
+LABEL org.opencontainers.image.description="Build image for Geany CI to support automatic building of Windows installers."
+LABEL org.opencontainers.image.url="https://github.com/geany/infrastructure"
+LABEL org.opencontainers.image.source="https://github.com/geany/infrastructure"
+LABEL org.opencontainers.image.authors="The Geany contributors"
+LABEL org.opencontainers.image.licenses="GPL-2.0"
+
+# install native tools and libraries
+RUN set -ex && \
+    dpkg --add-architecture i386 && \
+    apt-get update && \
+    apt-get install --no-install-recommends --assume-yes \
+    # libraries \
+    libcurl3-gnutls libgpgme11 libarchive13 libssl1.1 \
+    # common useful utilities \
+    wget curl less nano git gnupg2 file ca-certificates dos2unix \
+    zip unzip xz-utils zstd \
+    # build tools \
+    build-essential automake autoconf autopoint gettext libtool check cppcheck \
+    # genay-plugins autogen.sh requirements
+    intltool libglib2.0-dev \
+    # mingw-w64 \
+    gcc-mingw-w64-x86-64 g++-mingw-w64-x86-64 mingw-w64-x86-64-dev mingw-w64-tools \
+    # install wine to test installer and created binaries
+    wine wine32 wine64 \
+    # install NSIS and exiftool to inspect binary metadata
+    nsis libimage-exiftool-perl osslsigncode \
+    # Geany build dependencies \
+    python3-lxml python3-docutils
+
+
+# copy pacman and scripts
+COPY --from=build-pacman /windows /windows
+COPY --from=build-pacman /usr/local/pacman /usr/local/pacman
+COPY mingw64/bin/ /usr/local/bin/
+RUN ln -s /usr/local/pacman/bin/* /usr/local/bin/ && \
+    mkdir /build
+
+WORKDIR /build
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/pacman/lib/x86_64-linux-gnu
+
+# start wine to initially create config directory
+RUN /usr/local/bin/mingw-w64-i686-wine hostname.exe && \
+    /usr/local/bin/mingw-w64-x86_64-wine hostname.exe && \
+    # install GTK3 and all its dependencies
+    pacman --noconfirm -Sy mingw-w64-x86_64-gtk3 && \
+    # cleanup
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* && \
+    yes | pacman -Scc && \
```

Comment:
`--noconfirm`. `yes` doesn't terminate by itself which I is generally problematic IMO.

Reply:
I guess you did not test what you suggest?
`--noconfirm` does *not* clean the caches as the default for the question "Do you want to remove ALL files from cache? [y/N]" is no.

In this context `yes` seems to work as intended. Did you experience a certain problem?